crates/apub/src/protocol/objects/page.rs

```diff
@@ -124,6 +125,37 @@ impl Attachment {
       _ => None,
     }
   }
+
+  pub(crate) async fn as_markdown(&self, context: &Data<LemmyContext>) -> LemmyResult<String> {
+    let (url, name, is_image) = match self {
+      Attachment::Image(i) => (i.url.clone(), i.name.clone(), true),
+      Attachment::Document(d) => {
+        let is_image = d
+          .media_type
+          .as_ref()
+          .is_some_and(|media| media.starts_with("video") || media.starts_with("image"));
+
+        (d.url.clone(), d.name.clone(), is_image)
+      }
+      Attachment::Link(l) => {
+        let is_image = l
+          .media_type
+          .as_ref()
+          .is_some_and(|media| media.starts_with("video") || media.starts_with("image"));
+
+        (l.href.clone(), None, is_image)
+      }
+    };
+
+    if is_image {
+      let url = proxy_image_link_opt_apub(Some(url), context)
+        .await?
+        .expect("No url");
+      Ok(format!("![{}]({url})", name.unwrap_or_default()))
+    } else {
+      Ok(url.to_string())
```

Comment:
Also make a markdown link here, you can use the url as title (for clients which dont linkify plaintext links).